### DISCUSSION TITLE
How to add atributes and terms to affiliate products automatically while or after import?

### DISCUSSION THREAD

**Discussion by e-orlov** [datafeedr/wordpress-plugins#33] [Help]
I guess, this issue has multiple webmaster working with WooCommerce and Datafeedr. The issue: affiliate products are imported with product attributes from the products feed. These product data is limited and not very descriptive. I.e. in my verticale, from the whole product data coming with imported product I can use only name, price, boolean sale, salesprice, image link and description. But, beside of this, I have around 10 product attributes, and products in each category have an own set of attribute terms.

So my question: how can I `automatically` apply a special set of attribute terms to all products I import into certain category? `Automatically` is here enormously important - because product import works automatically and I don't know, when a new product was imported.

The only idea I have, is, to run a scheduled MySQL query in regular time intervals, like daily, and apply attribute terms to products, having specified category ID.

Do mayby somebody has any better idea?

**Comment by EricBusch** [comment]
Hi

Here's how you can automatically added attributes to your products: https://datafeedrapi.helpscoutdocs.com/article/221-import-attribute-for-a-product

Hope this helps!

If you have any other questions, please don't hesitate to ask.

Thanks,
Eric

**Comment by e-orlov** [comment]
Hello Eric,

I'm afraid, I'm not clearly expressed my need. The documentation you linked explains, how to import attributes and terms, which **are** part of the merchant's product data delivered with the feed.

My need is to apply attributes and terms, which are **not** delivered with product feed. Like this: I want, that all products I ever import with [this query](https://scratchpad.datafeedr.com?expires=1688309571&query=YToyOntzOjQ6Im5hbWUiO3M6MTA6InN1Ym1hcmluZXIiO3M6NToiYnJhbmQiO3M6NToidHVkb3IiO30%3D&signature=50527134b1b4c953bd5df9b88c9bacf71e4b0734a9c25928fe50532c7053ae83) will always have an attribute/term combination `case diameter: 41mm`, which is not included in the product data.

**Comment by EricBusch** [comment]
Hi

OK, I think I understand. We don't have a method of doing that but you may be able to modify this code so that it works with Attributes instead of Tags.

https://datafeedrapi.helpscoutdocs.com/article/257-import-custom-product-tags-for-each-product-set

Hope this helps!

Eric
